### ISSUE TITLE
Organization selection choices

### ISSUE THREAD

**Issue by Keyrxng** [ubiquity-os/ubiquity-os-plugin-installer#26]
> I am only seeing two orgs but I am in more orgs that that, how are they chosen?

Orgs to select for editing come from the below method currently

https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-the-authenticated-user

```ts
public async getGitHubUserOrgs(): Promise<string[]> {
    const octokit = await this.getOctokit();
    const response = await octokit.rest.orgs.listForAuthenticatedUser();
    return response.data.map((org: { login: string }) => org.login);
  }
```

1. Who is using this installer? I'd assume a redirect after https://github.com/ubiquity-os/ubiquity-os-kernel/issues/171 is complete to the installer UI which would naturally be the owner of the org. Then potentially who after that? Admin & Owner? Partner defined?

2. What orgs should be shown when a partner uses the installer?

- Orgs that they are an admin of or just owner of, collaborator of?
- Should the partner determine which roles can edit the config or do we enforce rules, if we enforce then what should they be?

We sort of scrapped global config variables from `ubiquity-os-config.yml` but we could leverage that to enforce partner defined roles which can edit (at least via the UI). 

E.g: The org owner must use it after `kernel-setup` and we allow them to input roles that can edit and we write this to the config separate from `plugins:` and we read this on fetch and check against our authenticated user.

_Originally posted by @Keyrxng in https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/25#issuecomment-2493797026_

**Comment by rndquu** [comment]
> Who is using this installer?

https://github.com/ubiquity-os/ubiquity-os-plugin-installer is used by partners who wants to start using the https://github.com/apps/ubiquity-os bot.

> I'd assume a redirect after https://github.com/ubiquity-os/ubiquity-os-kernel/issues/171 is complete to the installer UI

No, https://github.com/ubiquity-os/ubiquity-os-kernel/issues/171 is for collaborators. Partners should only interact with the https://github.com/ubiquity-os/ubiquity-os-plugin-installer, they don't need to deploy the bot from scratch, we've already installed it for them.

> Then potentially who after that? Admin & Owner? Partner defined?

Admin or owner, I suppose owner in most cases.

> What orgs should be shown when a partner uses the installer?

All orgs where authenticated user has a `write` access in the organization. If it's hard to get the access level via github API then all organizations where currently authenticated user is admin or owner (i.e. has a role with `write` access because that permission is required for editing the bot's config).

> E.g: The org owner must use it after kernel-setup and we allow them to input roles that can edit and we write this to the config separate from plugins: and we read this on fetch and check against our authenticated user.

Keep it simple for now, if authenticated user has `write` permissions then he's allowed to edit an organization config.

To sum up, as far as I understand, as a part of this issue we should fix [this](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2497319574) bug.

**Comment by Keyrxng** [comment]
> To sum up, as far as I understand, as a part of this issue we should fix https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2497319574 bug.

Yeah I think it deserves it's own task vs holding back https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25

> No, https://github.com/ubiquity-os/ubiquity-os-kernel/issues/171 is for collaborators.

Yes so it is that's my mistake, the redirect URL should be updated within the `UbiquityOS` GitHub app so that as soon as they install the app, they are redirected to the installer.

A redirect from `interactive-setup` would still make sense for contributors imo but I get it.

> Admin or owner, I suppose owner in most cases.
> All orgs where authenticated user has a write access in the organization. If it's hard to get the access level via github API then all organizations where currently authenticated user is admin or owner (i.e. has a role with write access because that permission is required for editing the bot's config).
> Keep it simple for now, if authenticated user has write permissions then he's allowed to edit an organization config.

Please see this [comment](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2500319084) but to sum up:

- `orgs.listForUser` is not reliable if the user has their membership as `private`
- extracting public orgs from above and trying to use `orgs.listMemberShipForAuthUser` throws the below error for `ubiquity` for my account, yet I am a `member` and I have `write` access because I can push straight to any repo.

![image](https://github.com/user-attachments/assets/c2220177-9cdd-4dfd-984d-1e6a88650608)

So it appears to me like org settings are affecting our API responses when using these methods (I tried with a PAT token and got the same results) which makes sense because `ubiquity` did not show for @gentlementlegen amongst others.

> I have seen only 2 out of 4 orgs where I am an admin with the rights to create a configuration (Meniole and ubiquity-os). I should have seen ubiquity-os-marketplace most likely.

**Comment by rndquu** [comment]
> > To sum up, as far as I understand, as a part of this issue we should fix [#25 (comment)](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2497319574) bug.
> 
> Yeah I think it deserves it's own task vs holding back #25
> 
> > No, [ubiquity-os/ubiquity-os-kernel#171](https://github.com/ubiquity-os/ubiquity-os-kernel/issues/171) is for collaborators.
> 
> Yes so it is that's my mistake, the redirect URL should be updated within the `UbiquityOS` GitHub app so that as soon as they install the app, they are redirected to the installer.
> 
> A redirect from `interactive-setup` would still make sense for contributors imo but I get it.
> 
> > Admin or owner, I suppose owner in most cases.
> > All orgs where authenticated user has a write access in the organization. If it's hard to get the access level via github API then all organizations where currently authenticated user is admin or owner (i.e. has a role with write access because that permission is required for editing the bot's config).
> > Keep it simple for now, if authenticated user has write permissions then he's allowed to edit an organization config.
> 
> Please see this [comment](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2500319084) but to sum up:
> 
> * `orgs.listForUser` is not reliable if the user has their membership as `private`
> * extracting public orgs from above and trying to use `orgs.listMemberShipForAuthUser` throws the below error for `ubiquity` for my account, yet I am a `member` and I have `write` access because I can push straight to any repo.
> 
> ![image](https://private-user-images.githubusercontent.com/106303466/389928087-c2220177-9cdd-4dfd-984d-1e6a88650608.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MjE1NDMsIm5iZiI6MTczMjcyMTI0MywicGF0aCI6Ii8xMDYzMDM0NjYvMzg5OTI4MDg3LWMyMjIwMTc3LTljZGQtNGRmZC05ODRkLTFlNmE4ODY1MDYwOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxNTI3MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mN2EwMDkxNDk4M2ViYjAyZDU2YzMyZDMzNTBkM2MxNTI1OTk2OGRiMDU0NWIxZmRkNzBjMDIxNzU0NTJjYmU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NMiEKPm8bgGUjdlBMN4cYkyV53vDOk1JpWazKabl__0)
> 
> So it appears to me like org settings are affecting our API responses when using these methods (I tried with a PAT token and got the same results) which makes sense because `ubiquity` did not show for @gentlementlegen amongst others.
> 
> > I have seen only 2 out of 4 orgs where I am an admin with the rights to create a configuration (Meniole and ubiquity-os). I should have seen ubiquity-os-marketplace most likely.

So, as far as I understand from [this](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25#issuecomment-2500302845) comment, if user's membership is private then organization won't be displayed on plugin setup. Is there a way (i.e. API method) to get all user organizations (i.e. with both public and private organization membership)?

**Comment by Keyrxng** [comment]
@rndquu #31 

I tried near all available methods and all of them errored when I tried to find out more about their permissions (via the error screenshot above) through `rest.org` although `rest.repo` allows you to check permission for a particular repo which seems to bypass whatever is preventing the `rest.org`.

So if `-marketplace` shows up for you guys then the PR fixes things I think

**Comment by rndquu** [comment]
@gentlementlegen In this github issue task reward is 75 USD while task solver received 150 WXDAI. Is it a bug or somewhat expected?

**Comment by 0x4007** [comment]
Limits are enforced for all rewards to the assignee task reward amount. The warning exists in the breakdown. 

Currently the only way to go beyond the limit is if the assignee wrote the spec as well. 

Perhaps we need to think through this more but to me it makes sense that if the assignee also wrote the spec that they should receive more rewards. 

Lastly I still believe that we need to dial down spec rewards, perhaps remove the priority multiplier on it.

**Comment by gentlementlegen** [comment]
Right now you can get maximum twice the total of the reward, so in this case 75 for the reward, plus 75 for incentives (comments, spec etc) totaling 150. I didn't intend that total at first but it was deemed as a feature so for now it works that way, we can always change it if needed.

@0x4007 all it takes is setting this to one in the cfg: https://github.com/ubiquity-os/.ubiquity-os/blob/main/.github/.ubiquity-os.config.dev.yml#L67

**Comment by 0x4007** [comment]
As I understand you're doing a multiplier and then multiplying the priority level so 3 x 4 =12x multiplier which is too high.

**Comment by gentlementlegen** [comment]
That is correct, first multiplier is applied to the comment evaluation and the priority then applies to the final result so x12. Putting the multiplier back to 1 should solve that.
